ElementX/Sources/Services/Client/ClientProxy.swift

```diff
@@ -141,6 +141,31 @@ class ClientProxy: ClientProxyProtocol {
         slidingSyncObserverToken = nil
     }
     
+    func directRoomForUserID(_ userID: String) async -> Result<String?, ClientProxyError> {
+        await Task.dispatch(on: clientQueue) {
+            do {
+                if let room = try self.client.getDmRoom(userId: userID) {
+                    return .success(room.id())
+                }
+            } catch {
+                return .failure(.failedRetrievingDirectRoom)
+            }
+            return .success(nil)
+        }
+    }
```

Comment:
We can make it a bit more readable
```suggestion
    func directRoomForUserID(_ userID: String) async -> Result<String?, ClientProxyError> {
        await Task.dispatch(on: clientQueue) {
            do {
                let room = try self.client.getDmRoom(userId: userID)?.id()
                return room.map { .success($0) } ?? .success(nil)
            } catch {
                return .failure(.failedRetrievingDirectRoom)
            }
        }
    }
```